File: backend/src/db/mongo/models/Document.ts
import { Schema, model, Document as MongooseDocument } from 'mongoose';

export interface IDocument extends MongooseDocument {
    name: string;
    documentId: string;
    userId: string;
    uploadDate: string;
    s3documentId: string;
    activityGenerationComplete: boolean;
    public: boolean;
}

const documentSchema = new Schema<IDocument>({
    name: { type: String, required: true },
    userId: { type: String, required: true },
    uploadDate: { type: String, required: true },
    s3documentId: { type: String, required: true, unique: true },
    documentId: { type: String, required: true },
    activityGenerationComplete: {
        type: Boolean,
        required: true,
        default: false,
    },
    public: {
        type: Boolean,
        default: false,
    },
});

const Document = model<IDocument>('Document', documentSchema);

export default Document;

File: frontend/app/src/main/java/com/example/thinkr/data/models/Document.kt
package com.example.thinkr.data.models

import kotlinx.serialization.Serializable
import kotlinx.serialization.Transient

/**
 * Data model representing a document in the system.
 *
 * Contains information about a document including its identifier, name, upload status,
 * and activity generation status. Includes transient properties for UI state management.
 *
 * @property documentId The unique identifier assigned to the document.
 * @property documentName The name of the document.
 * @property uploadTime The timestamp when the document was uploaded.
 * @property activityGenerationComplete Boolean indicating whether learning activities have been generated for this document.
 * @property isUploading Transient property indicating whether the document is currently being uploaded.
 * @property documentContext Transient property containing additional context or content for the document.
 */
@Serializable
data class Document(
    val documentId: String,
    val documentName: String,
    val uploadTime: String,
    val activityGenerationComplete: Boolean,
    val public: Boolean,
    @Transient
    val isUploading: Boolean = false,
    @Transient
    val documentContext: String = ""
)
